state/allwatcher_internal_test.go

```diff
@@ -2473,8 +2473,7 @@ func testChangeUnits(c *gc.C, owner names.UserTag, runChangeTests func(*gc.C, []
 			c.Assert(err, jc.ErrorIsNil)
 			err = u.AssignToMachine(m)
 			c.Assert(err, jc.ErrorIsNil)
-			err = u.OpenPorts("tcp", 21, 22)
-			c.Assert(err, jc.ErrorIsNil)
+			fw.AssertOpenUnitPorts(c, u, emptySubnet, "tcp", 21, 22)
```

Comment:
I do like these, but splitting them out into their own patch would have made it quite a bit easier to review. :)

Reply:
Unfortunately that was not possible as the changes basically broke a lot of tests :-(